### PR TITLE
Cleanup document code

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Not exhaustive:
- Drop unused `Document::uploadDocument()`
- PHPDoc fixes
- Some minor DB query optimizations
- Use self/static for static calls
- Use `::class` instead of `getType` and hard-coded class names

### CODE REVIEW

File: src/Document.php [hunk 1]
```diff
@@ -158,29 +126,27 @@ public function canCreateItem()
...
 
-       // UNLINK DU FICHIER
+       // Unlink/delete the file
         if (!empty($this->fields["filepath"])) {
             if (
                 is_file(GLPI_DOC_DIR . "/" . $this->fields["filepath"])
                 && !is_dir(GLPI_DOC_DIR . "/" . $this->fields["filepath"])
                 && (countElementsInTable(
-                    $this->getTable(),
+                    self::getTable(),
```
Comment (trasher) [inline src/Document.php:149]: I don't know if anything currently override Document class, but replacing `$this` should probably be replaced with `static::` rather  than `self::` the scope is not the same

File: src/Document_Item.php [hunk 2]
```diff
@@ -109,17 +104,17 @@ public function prepareInputForAdd($input)
             return false;
         }
 
-       // Do not insert circular link for document
+        // Do not insert circular link for document
         if (
-            ($input['itemtype'] == 'Document')
-            && ($input['items_id'] == $input['documents_id'])
+            ($input['itemtype'] === 'Document')
```
Comment (trasher) [inline src/Document_Item.php:109]:
```suggestion
            ($input['itemtype'] === Document::class)
```

File: src/Document_Item.php [hunk 3]
```diff
@@ -208,8 +197,7 @@ public function pre_deleteItem()
 
     public function post_addItem()
     {
-
-        if ($this->fields['itemtype'] == 'Ticket') {
+        if ($this->fields['itemtype'] === 'Ticket') {
```
Comment (trasher) [inline src/Document_Item.php:200]:
```suggestion
        if ($this->fields['itemtype'] === Ticket::class) {
```

File: src/Document_Item.php [hunk 4]
```diff
@@ -228,15 +216,9 @@ public function post_addItem()
...
-    /**
-     * @since 0.83
-     *
-     * @see CommonDBTM::post_purgeItem()
-     **/
     public function post_purgeItem()
     {
-
-        if ($this->fields['itemtype'] == 'Ticket') {
+        if ($this->fields['itemtype'] === 'Ticket') {
```
Comment (trasher) [inline src/Document_Item.php:221]:
```suggestion
        if ($this->fields['itemtype'] === Ticket::class) {
```

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**